src/workspace/mod.rs

```diff
@@ -2145,7 +2146,93 @@ mod test {
 
   use super::*;
 
-  struct UnreachableSys;
+  #[derive(Debug)]
+  pub struct UnreachableMetadata;
+
+  impl sys_traits::FsMetadataValue for UnreachableMetadata {
+    fn file_type(&self) -> sys_traits::FileType {
+      unreachable!()
+    }
+    fn len(&self) -> u64 {
+      unreachable!()
+    }
+    fn accessed(&self) -> std::io::Result<std::time::SystemTime> {
+      unreachable!()
+    }
+    fn created(&self) -> std::io::Result<std::time::SystemTime> {
+      unreachable!()
+    }
+    fn changed(&self) -> std::io::Result<std::time::SystemTime> {
+      unreachable!()
+    }
+    fn modified(&self) -> std::io::Result<std::time::SystemTime> {
+      unreachable!()
+    }
+    fn dev(&self) -> std::io::Result<u64> {
+      unreachable!()
+    }
+    fn ino(&self) -> std::io::Result<u64> {
+      unreachable!()
+    }
+    fn mode(&self) -> std::io::Result<u32> {
+      unreachable!()
+    }
+    fn nlink(&self) -> std::io::Result<u64> {
+      unreachable!()
+    }
+    fn uid(&self) -> std::io::Result<u32> {
+      unreachable!()
+    }
+    fn gid(&self) -> std::io::Result<u32> {
+      unreachable!()
+    }
+    fn rdev(&self) -> std::io::Result<u64> {
+      unreachable!()
+    }
+    fn blksize(&self) -> std::io::Result<u64> {
+      unreachable!()
+    }
+    fn blocks(&self) -> std::io::Result<u64> {
+      unreachable!()
+    }
+    fn is_block_device(&self) -> std::io::Result<bool> {
+      unreachable!()
+    }
+    fn is_char_device(&self) -> std::io::Result<bool> {
+      unreachable!()
+    }
+    fn is_fifo(&self) -> std::io::Result<bool> {
+      unreachable!()
+    }
+    fn is_socket(&self) -> std::io::Result<bool> {
+      unreachable!()
+    }
+    fn file_attributes(&self) -> std::io::Result<u32> {
+      unreachable!()
+    }
+  }
+
+  pub struct UnreachableSys;
+
+  impl sys_traits::BaseFsMetadata for UnreachableSys {
+    type Metadata = UnreachableMetadata;
```

Comment:
Maybe just have this be `std::fs::Metadata` so you don't need the extra `UnreachableMetadata`?